### PR TITLE
Peds, Hud, CFO 1/2, fixes - including zone names

### PR DESCRIPTION
**Pull request by erorcun**
All of the fixes/refactorings in here is already in master.

Sadly zone names will continue to be broken on current save games, but can be fixed easily (see Zones.cpp)

### CODE REVIEW

File: src/peds/PlayerPed.cpp [hunk 1]
```diff
@@ -1828,6 +1827,100 @@ CPlayerPed::RemovePedFromMeleeList(CPed *ped)
...
+	}
+
+	// We don't have victim ped in our melee list
+	if (dirToAttack == -1 && thereIsAnEmptySlot) {
+		float angle = Atan2(-dist.x, -dist.y);
+		float adjustedAngle = angle + DEGTORAD(30.0f);
+		if (adjustedAngle < 0.f)
+			adjustedAngle += TWOPI;
+
+		int wantedDir = floorf(adjustedAngle / DEGTORAD(60.0f));
```
Comment (ShFil119) [inline src/peds/PlayerPed.cpp:1891]: new wrapper

File: src/peds/Ped.cpp [hunk 2]
```diff
@@ -20547,6 +20360,536 @@ CPed::PositionAttachedPed()
...
+					SetIdle();
+				}
+			} else {
+				if (m_nPedState != PED_SEEK_ENTITY && m_nPedState != PED_SEEK_POS
+					&& !bStopAndShoot && !killPlayerInNoPoliceZone && !bKindaStayInSamePlace) {
+					Say(SOUND_PED_ATTACK);
+					SetSeek(m_pedInObjective, wepRangeAdjusted);
+					bIsRunning = true;
+					if (m_nPedType == PEDTYPE_COP && FindPlayerPed()->m_pWanted->m_CurrentCops > 1) {
+						if (CGeneral::GetRandomNumber() & 1)
```
Comment (ShFil119) [inline src/peds/Ped.cpp:20877]: I believe we have something dedicated. randomtrueorfalse(?)

Comment (erorcun) [reply src/peds/Ped.cpp:20877]: it checks for `GetRandomNumber() < MYRAND_MAX / 2`

File: src/peds/PlayerPed.cpp (hunk 1 above)
Comment (erorcun) [reply src/peds/PlayerPed.cpp]: done

**Pull request merged**